src/main/java/com/minecolonies/coremod/client/gui/WindowResearchTree.java

```diff
@@ -137,299 +199,775 @@ public void onButtonClicked(@NotNull final Button button)
-            if (depth != 0 && abandoned == false && state == ResearchState.NOT_STARTED && parentResearch.hasResearchedChild(tree) && parentResearch.hasOnlyChild())
+            final boolean trueAbandoned = drawResearchItem(view, offsetX, offsetY, research, abandoned);
+
+            if (!research.getParent().isEmpty())
             {
-                trueAbandoned = true;
+                drawArrows(view, offsetX + INITIAL_X_OFFSET - X_SPACING, offsetY, researchList, research.getParent(), i, nextHeight, height);
             }
 
-            final Gradient gradient = new Gradient();
-            gradient.setSize(GRADIENT_WIDTH, GRADIENT_HEIGHT);
-            gradient.setPosition(offsetX + INITIAL_X_OFFSET, (nextHeight + Math.min(i, 1)) * (GRADIENT_HEIGHT + Y_SPACING) + Y_SPACING);
-            if (state == ResearchState.IN_PROGRESS)
+            if (!research.getChildren().isEmpty())
             {
-                gradient.setGradientStart(227, 249, 184, 255);
-                gradient.setGradientEnd(227, 249, 184, 255);
-                view.addChild(gradient);
+                nextHeight =
+                  drawTree(nextHeight + Math.min(i, 1), depth + 1, view, research.getChildren(), trueAbandoned);
             }
-            else if (trueAbandoned && state != ResearchState.FINISHED)
+            else
             {
-                gradient.setGradientStart(191, 184, 172, 255);
-                gradient.setGradientEnd(191, 184, 172, 255);
-                view.addChild(gradient);
+                nextHeight += Math.min(i, 1);
             }
-            else if (!parentResearched)
+        }
+
+        return nextHeight;
+    }
+
+    /**
+     * Draw the background gradients and labels for the research tree.
+     *
+     * @param view the view to append it to.
+     */
+    private void drawTreeBackground(final ZoomDragView view)
+    {
+        for (int i = 1; i < MAX_DEPTH; i++)
+        {
+            if (building.getBuildingLevel() < i)
             {
-                gradient.setGradientStart(239, 230, 215, 255);
-                gradient.setGradientEnd(239, 230, 215, 255);
+                final Gradient gradient = new Gradient();
+                gradient.setGradientStart(80, 80, 80, 70);
+                gradient.setGradientEnd(60, 60, 60, 70);
+                gradient.setSize(GRADIENT_WIDTH + X_SPACING, 1200);
+                gradient.setPosition((i - 1) * (GRADIENT_WIDTH + X_SPACING), 0);
                 view.addChild(gradient);
             }
-            else if (state != ResearchState.FINISHED)
+        }
+
+        if (building.getBuildingLevel() < building.getBuildingMaxLevel() || hasMax)
+        {
+            final Gradient gradient = new Gradient();
+            gradient.setGradientStart(80, 80, 80, 70);
+            gradient.setGradientEnd(60, 60, 60, 70);
+            gradient.setSize(GRADIENT_WIDTH + X_SPACING, 1200);
+            gradient.setPosition((6 - 1) * (GRADIENT_WIDTH + X_SPACING), 0);
```

Comment:
```suggestion
            gradient.setPosition((MAX_DEPTH - 1) * (GRADIENT_WIDTH + X_SPACING), 0);
```
only if you are not gonna rewrite it